target/linux/ath79/image/generic.mk

```diff
@@ -2229,6 +2236,31 @@ define Device/zbtlink_zbt-wd323
 endef
 TARGET_DEVICES += zbtlink_zbt-wd323
 
+define Device/zyxel_nwa112x
+  SOC := ar9342
+  DEVICE_VENDOR := ZyXEL
+  LOADER_TYPE := bin
+  LOADER_FLASH_OFFS := 0x050000
+  COMPILE := loader-$(1).bin loader-$(1).uImage
+  COMPILE/loader-$(1).bin := loader-okli-compile
+  COMPILE/loader-$(1).uImage := append-loader-okli $(1) | pad-to 64k | \
+	lzma | uImage lzma
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49
+  IMAGE_SIZE := 8192k
+  IMAGES += factory-$$$$(ZYXEL_MODEL_STRING).bin
+  IMAGE/factory-$$$$(ZYXEL_MODEL_STRING).bin := \
+	$$(IMAGE/default) | pad-to 8192k | check-size | zyxel-tar-bz2 \
```

Comment:
I don't think IMAGE/default is defined here?

Reply:
Does this imply the value is currently taken from `Device/wd_mynet-n750`, which has it declared via `Device/seama`, thus (accidentally) creating a valid image?
i.e. this would break as soon as a new device is introduced in between the two, that overrides this with an incompatible definition? Interesting, I had initially considered this a boilerplate defined somewhere in the upper Makefile structures, will fix by the start of next week, thanks.

Reply:
I'm not quite sure. If per-device variables have no default value (e.g. in Device/Default), they indeed do take the last value set to them. The question is whether you tested factory flashing after you replaced the IMAGE/sysupgrade here, since the padding will produce an image with reasonable size even if IMAGE/default would just do nothing.

Anyway, just put the correct individual commands here ...